File: info/snippets/old/conturdetection.cpp
#include <opencv2/opencv.hpp>

int main()
{
    // Bild laden
    cv::Mat image = cv::imread("bild.jpg", cv::IMREAD_COLOR);
    
    if (image.empty())
    {
        std::cout << "Konnte Bild nicht laden!" << std::endl;
        return -1;
    }

    // Bild in Graustufen konvertieren
    cv::Mat gray;
    cv::cvtColor(image, gray, cv::COLOR_BGR2GRAY);

    // Bildschwellenwertung // Gelb
    cv::Mat hsv;
    cv::cvtColor(image, hsv, cv::COLOR_BGR2HSV);
    cv::Scalar lower_yellow(0, 100, 100);
    cv::Scalar upper_yellow(30, 255, 255);
    cv::inRange(hsv, lower_yellow, upper_yellow, gray);
    cv::threshold(gray, gray, 200, 255, cv::THRESH_BINARY);

    //weiß
    //cv::threshold(gray, gray, 200, 255, cv::THRESH_BINARY);


    // Konturerkennung
    std::vector<std::vector<cv::Point>> contours;
    std::vector<cv::Vec4i> hierarchy;
    cv::findContours(gray, contours, hierarchy, cv::RETR_EXTERNAL, cv::CHAIN_APPROX_SIMPLE);

    // Leeres Bild für die Konturen erstellen
    cv::Mat contoursImage = cv::Mat::zeros(image.size(), CV_8UC1);

    // Konturen im Konturenbild zeichnen
    for (size_t i = 0; i < contours.size(); i++)
    {
        cv::drawContours(contoursImage, contours, static_cast<int>(i), cv::Scalar(255), cv::FILLED);
    }

    // Ergebnisbild anzeigen
    cv::imshow("Konturerkennung", contoursImage);
    cv::waitKey(0);

    return 0;
}

File: info/snippets/old/conturdetection.py
import cv2
import numpy as np

# Laden des Bildes
image = cv2.imread('panorama.png')

# Überprüfen, ob das Bild erfolgreich geladen wurde
if image is None:
    print("Fehler: Das Bild konnte nicht geladen werden.")
    exit()

# Konvertieren in Graustufen
gray = cv2.cvtColor(image, cv2.COLOR_BGR2GRAY)

# Anwenden einer Schwellenwertbildung
_, thresholded = cv2.threshold(gray, 0, 255, cv2.THRESH_BINARY_INV+cv2.THRESH_OTSU)

# Konturen finden
contours, hierarchy = cv2.findContours(thresholded, cv2.RETR_LIST, cv2.CHAIN_APPROX_SIMPLE)

# Einen leeren weißen Hintergrund erstellen
result = np.full_like(image, (255, 255, 255), dtype=np.uint8)

# Einen leeren schwarzen Hintergrund erstellen
#result =  np.zeros_like(image)

# Konturen zeichnen, die nahezu geschlossen sind

#for contour in contours:
#    epsilon = 0.1 * cv2.arcLength(contour, True)
#    approx = cv2.approxPolyDP(contour, epsilon, True)
#    if cv2.isContourConvex(approx):
#        cv2.drawContours(result, [contour], -1, (1, 1, 1), cv2.FILLED)

# Nur geschlossene Konturen in Weiß zeichnen
for contour in contours:
    if cv2.isContourConvex(contour):
        cv2.drawContours(result, [contour], -1, (1, 1, 1), cv2.FILLED)


# Ergebnis anzeigen
cv2.imshow('Nearly Closed Contours in Black', result)

# Ergebnis speichern
cv2.imwrite('panorama_contours.png', result)

cv2.waitKey(0)
cv2.destroyAllWindows()